File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root) {
        let stack = new Array();
        let cur = root;
        while (cur || stack.length > 0){
            while ( cur ){
                stack.push(cur);
                cur = cur.left;
            }
            cur = stack.pop();
            let temp = cur.right;
            cur.right = cur.left
            cur.left = temp;
            cur = cur.left;
        }
    return root;
    }
     
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        stack = []
        cur = root
        t = 1
        while cur or stack:
            while cur:
                stack.append(cur)
                cur = cur.left
            cur = stack.pop()
            if k == t:
                return cur.val
            cur = cur.right
            t += 1
        return None